src/task.ts

```diff
@@ -6,8 +6,21 @@ export type Quest<T extends Task = Task> = {
   tasks: T[];
 };
 
+export type OutfitSlot =
+  | "hat"
+  | "back"
+  | "weapon"
+  | "offhand"
+  | "shirt"
+  | "pants"
+  | "acc1"
+  | "acc2"
+  | "acc3"
+  | "familiar";
+
 export interface OutfitSpec {
-  equip?: Item[]; // Items to be equipped in any slot
+  items?: Item[]; // Items to be equipped in any slot
+  equip?: Partial<{ [slot in OutfitSlot]: Item | Item[] }>; // Items to be equipped in specific slots
```

Comment:
I would keep equip as an Item[] and add a new guy for the new thing. For backwards compatibility.

I would also articulate this point better.